apple/internal/testing/build_test_rules.bzl

```diff
@@ -121,9 +121,13 @@ number (for example, `"9.0"`).
             ),
         },
         doc = doc,
-        exec_compatible_with = [
-            "@platforms//os:macos",
-        ],
+        exec_groups = {
+            "test": exec_group(
+                exec_compatible_with = [
+                    "@platforms//os:macos",
+                ],
+            ),
+        },
```

Comment:
Does this affect `transition_support.apple_rule_transition`? The main reason for `exec_compatible_with` was to make sure the transition saw the right `cpu`, so it could assign the right default. Without this, it was looking at the host platform, which in the case of Linux, would mean x86_64 (usually) instead of the correct arm64.